components/env.rst

```diff
@@ -0,0 +1,114 @@
+.. index::
+   single: Env
+   single: Components; Env
+
+The Env Component
+=================
+
+    The Env Component parses ``.env`` files to make environment variables
+    stored in them accessible via ``getenv()``, ``$_ENV``, or ``$_SERVER``.
+
+Installation
+------------
+
+You can install the component in 2 different ways:
+
+* :doc:`Install it via Composer </components/using_components>` (``symfony/env`` on `Packagist`_);
+* Use the official Git repository (https://github.com/symfony/env).
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+Sensitive information and environment-dependent settings should be defined as
+environment variables (as recommended for `twelve-factor applications
+<http://www.12factor.net/>`_. Using an `.env` file to store those environment
+variables eases development and CI management by keeping them in one "standard"
+place and agnostic of the technology stack you are using (Nginx vs PHP build-in
+server for instance).
+
+.. note::
+
+    PHP has a lot of different implementation of this "pattern". This
+    implementation goal is to replicate what ``source .env`` would do. So, it
+    tries to be as similar as possible with the default shells behavior, and
+    does not do anything not possible via ``source`` (like value validation).
+
+Usage
+-----
+
+First, require Symfony Env via Composer:
+
+.. code-block::
+
+    composer require symfony/env
+
+Load an ``.env`` file in your PHP application via ``Dotenv::load()``:
+
+.. code-block:: php
+
+    use Symfony\Component\Env\Dotenv;
+
+    $dotenv = new Dotenv();
+    $dotenv->load(__DIR__.'/.env');
+
+    // You can also load several files
+    $dotenv->load(__DIR__.'/.env', __DIR__.'/.env.dev');
+
+Given the following ``.env`` file content:
+
+.. code-block:: bash
+
+    DB_USER=root
+    DB_PASS=pass
+
+Access the value with ``getenv()`` in your code:
+
+.. code-block:: php
+
+    $dbUser = getenv('DB_USER');
+    // you can also use ``$_ENV`` or ``$_SERVER``
+
+.. note::
+
+    Symfony Env never overwrites existing environment variables.
+
+You should never store a ``.env`` file is your code repository as it might
+contains sensitive information; creates a ``.env.dist`` file with sensible
+defaults instead.
+
+Symfony Env should only be used in development/testing/staging environments.
```

Comment:
This recommendation contradicts a bit the previous one. If this file should only be used for non-production environments, then there's no problem storing this file in the repo because the credentials will be different. If this is true, then we don't need `.env.dist` and we can simplify things a lot.